src/Storages/ObjectStorageQueue/ObjectStorageQueueOrderedFileMetadata.cpp

```diff
@@ -642,4 +774,12 @@ void ObjectStorageQueueOrderedFileMetadata::filterOutProcessedAndFailed(
     paths = std::move(result);
 }
 
+std::string ObjectStorageQueueOrderedFileMetadata::getHivePart(const std::string & file_path)
+{
+    std::string hive_part(VirtualColumnUtils::findHivePartitioningInPath(file_path));
+    boost::replace_all(hive_part, "_", "__");
+    boost::replace_all(hive_part, "/", "_");
```

Comment:
As I see we do this as we use it as a node name in keeper so we want to kind of normalize it, right? The method name implies that we return hive part as is without normalization, so let's split it into two methods to make it more explicit and understandable